src/Middleware/Diagnostics/src/ExceptionHandler/ExceptionHandlerMiddlewareImpl.cs

```diff
@@ -48,7 +48,7 @@ internal sealed class ExceptionHandlerMiddlewareImpl
         _metrics = new DiagnosticsMetrics(meterFactory);
         _problemDetailsService = problemDetailsService;
 
-        if (_options.ExceptionHandler == null)
+        if (_options.ExceptionHandler == null && _exceptionHandlers.Length == 0)
```

Comment:
I don't understand why we'd ever set `_options.ExceptionHandler = _next`.  Isn't `_next` the thing that might throw, necessitating handling?

Reply:
It looks like the only consumer does a null check first, so it's not to guarantee a non-null value.

Reply:
It's set to _next when we also re-write the path and re-route the request to something that should be able to handle the exception.

Reply:
Sorry, I'm not following.  It looks like `_next` is invoked first thing in `Invoke` and all the handling is downstream.  Doesn't that mean `_next` has already been invoked by the time `_options.ExceptionHandler` has been invoked?  Maybe you're saying that we rewrite the path _after_ the exception and then intentionally invoke `_next` _again_?

Reply:
Yes :-)
https://github.com/dotnet/aspnetcore/blob/b71c6d0295341ec354fe44d24f31de3f6fb79e2e/src/Middleware/Diagnostics/src/ExceptionHandler/ExceptionHandlerMiddlewareImpl.cs#L142-L187

Reply:
That seems like the right place to update the handler?

Reply:
? The Options shouldn't be modified after the constructor.

Reply:
If that's the convention, then fair enough.